default.nix

```diff
@@ -345,6 +345,23 @@ let overrideCabal = pkg: f: if pkg == null then null else haskellLib.overrideCab
             };
             sha256 = "1gx9p9i5jli91dnvvrc30j04h1v2m3d71i8sxli6qrhplq5y63dk";
           }) {};
+
+        ########################################################################
+        # Convenience functions
+        ########################################################################
+
+        # Variant of `callCabal2nix` optimized for local
+        # packages. Specifically, it only uses `cabal2nix` on the
+        # cabal file, so that the "import-from-derivation" is only
+        # recomputed when the cabal file changes, and so your source
+        # code isn't duplicated into the nix store on every change.
+        # The drawback is that it only works on Nix path literals, not
+        # string paths.
```

Comment:
Beautiful!  It seems like we should use this everywhere we can :)  I wonder if it's possible to change callCabal2nix to use this when appropriate.